op_join.go

```diff
@@ -23,9 +24,66 @@ func (JoinOperator) Phase() OperatorPhase {
 	return EvalPhase
 }
 
-// Dependencies ...
-func (JoinOperator) Dependencies(_ *Evaluator, _ []*Expr, _ []*tree.Cursor) []*tree.Cursor {
-	return []*tree.Cursor{}
+// Dependencies returns the nodes that (( join ... )) requires to be resolved
+// before its evaluation. Returns no dependencies on error, because who cares
+// about eval order if Run is going to bomb out anyway.
+func (JoinOperator) Dependencies(ev *Evaluator, args []*Expr, _ []*tree.Cursor) []*tree.Cursor {
+	DEBUG("Calculating dependencies for (( join ... ))")
+	deps := []*tree.Cursor{}
+	if len(args) < 2 {
+		DEBUG("Not enough arguments to (( join ... ))")
+		return []*tree.Cursor{}
+	}
+
+	//skip the separator arg
+	for _, arg := range args[1:] {
+		if arg.Type == Literal {
+			continue
+		}
+		if arg.Type != Reference {
+			DEBUG("(( join ... )) argument not Literal or Reference type")
+			return []*tree.Cursor{}
```

Comment:
Seems like this wouldn't handle `(( join "" "http://" meta.server_list ))` properly, since it returns no dependencies after the first non-reference argument? I didn't see any tests that would exercise this, but might have missed them